### PR TITLE
Cosign policy checker

### PR DESCRIPTION
**Pull request by wy65701436**
Enable policy checker for cosign, when it's enabled, user cannot pull artifact without cosign.

Signed-off-by: wang yan <wangyan@vmware.com>

### CODE REVIEW

**Review by chlins — APPROVED** [review]
lgtm

File: src/server/middleware/contenttrust/notaryv1.go [hunk 1]
```diff
@@ -28,8 +28,8 @@ var (
 	}
 )
 
-// Middleware handle docker pull content trust check
-func Middleware() func(http.Handler) http.Handler {
+// NotaryV1 handle docker pull content trust check
+func NotaryV1() func(http.Handler) http.Handler {
```
Comment (heww) [inline src/server/middleware/contenttrust/notaryv1.go:32]: is there version 2 of the notary? If not I think it's enough to name it as `Notary` here.

Comment (wy65701436) [reply src/server/middleware/contenttrust/notaryv1.go:32]: yes, there is. https://github.com/notaryproject/notaryproject

File: src/server/middleware/util/util.go [hunk 2]
```diff
@@ -57,13 +56,17 @@ func ParseProjectName(r *http.Request) string {
...
-	// only scanner pull access can bypass.
-	if ok && secCtx.Name() == "v2token" &&
-		secCtx.Can(ctx, rbac.ActionScannerPull, project.NewNamespace(projectID).Resource(rbac.ResourceRepository)) {
-		return true
+	// 1, scanner pull access can bypass.
+	// 2, cosign pull can bypass, it needs to pull the manifest before pushing the signature.
+	if ok && secCtx.Name() == "v2token" {
+		if secCtx.Can(r.Context(), rbac.ActionScannerPull, project.NewNamespace(projectID).Resource(rbac.ResourceRepository)) ||
+			(secCtx.Can(r.Context(), rbac.ActionPush, project.NewNamespace(projectID).Resource(rbac.ResourceRepository)) &&
+				strings.Contains(r.UserAgent(), "cosign")) {
```
Comment (heww) [inline src/server/middleware/util/util.go:67]: Is there an escape here?

I think we only can skip checking the sig image, not all the images.

File: src/server/middleware/contenttrust/notaryv1.go (hunk 1 above)
Comment (wy65701436) [reply src/server/middleware/contenttrust/notaryv1.go:32]: it's ok to use notary.

**Review by heww — APPROVED** [review]

File: src/server/middleware/util/util.go (hunk 2 above)
Comment (wy65701436) [reply src/server/middleware/util/util.go:67]: yes, the request from cosign client with push permission can bypass the check.

**Pull request merged**